File: remind_me_be/RemindMe.ToDo/RemindMe.ToDo.Application/Requests/ToDos/CreateToDoRequest.cs
using MediatR;
using RemindMe.Domain.Entities;

namespace RemindMe.Application.Requests.Todos
{
    public class CreateTodoRequest : IRequest<Item>
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public string CreationDate { get; set; }
        public string StartDate { get; set; }
        public string EndDate { get; set; }
        public bool IsFinished { get; set; }
        public int Difficulty { get; set; }
        public int OwnerId { get; set; }

    }
}

File: remind_me_fe/lib/features/todos/data/models/create_todo_request.dart
class CreateToDoRequest {
  String title;
  String description;
  DateTime creationDate;
  DateTime startDate;
  DateTime? endDate;
  bool isFinished;
  int difficulty;
  String ownerId;

  CreateToDoRequest({
    required this.title,
    required this.description,
    required this.creationDate,
    required this.startDate,
    this.endDate,
    required this.isFinished,
    required this.difficulty,
    required this.ownerId,
  });

  Map<String, dynamic> toJson() {
    Map<String, dynamic> result = {
      'title': title,
      'description': description,
      'creationDate': creationDate.toIso8601String(),
      'startDate': startDate.toIso8601String(),
      'endDate': endDate?.toIso8601String(),
      'isFinished': isFinished,
      'difficulty': difficulty,
      'ownerId': ownerId,
    };
    return result;
  }
}
